API/Repositories/Implementations/MatchesRepository.cs

```diff
@@ -38,6 +38,13 @@ public async Task<Match> UpdateVerificationStatus(int id, int? verificationStatu
         return existing;
     }
 
+    public async Task<Match?> GetByNameAsync(string name)
```

Comment:
Stage's comment is still relevant. I think `SearchAsync` is more proper since the result wouldn't fit as an overload to `GetAsync`. Adjust this where present.